opm/autodiff/BlackoilWellModel_impl.hpp

```diff
@@ -140,7 +157,17 @@ namespace Opm {
         // update the previous well state. This is used to restart failed steps.
         previous_well_state_ = well_state_;
 
-
+        // Do the well testing if enabled
+        if (!initial_step_ && wtest_config.size() > 0 && enableWellTests) {
+            // solve the well equation isolated from the reservoir.
+            assemble(/*iterationIdx=*/ 0, /*dt (not relevant for well test) =*/ 1.0, enableWellTests);
+            // This is where WECON is checked
+#warning WECON should be tested after every time step not after every report step
+            endReportStep();
```

Comment:
@GitPaean The current Flow only test for WECON after every report step, shouldn't it be every time-step? Unfortunately, it is not-trivial to fix this since the WECON dynamic_list_econ_limited_ is passed and managed by the well manager.

Reply:
I do not recall the details, please check the manual for `WECON` and make the corresponding change. 

Because there will be multiple possible reasons to shut a well, we need to maintains some data structure in WellInterface to help `WTEST`. And shut wells need to be in well_containers_ to perform WTEST, so we need to have a flag for the WellInterface to indicate if the well shut for some reasons or open. 

> Unfortunately, it is not-trivial to fix this since the WECON dynamic_list_econ_limited_ is passed and managed by the well manager.

The WECON was implemented without knowledge about WTEST, I think very likely dynamic_list_econ_limited related needs to be moved to BlackoilWellModel.  It is not easy indeed.  Thanks a lot for helping out about this keyword.

Reply:
> Because there will be multiple possible reasons to shut a well, we need to maintains some data structure in WellInterface to help WTEST.

That is the purpose of this: https://github.com/OPM/opm-common/pull/414